var/spack/repos/builtin/packages/openssl/package.py

```diff
@@ -17,6 +21,42 @@ class Openssl(Package):
     depends_on("zlib")
     parallel = False
 
+    def url_for_version(self, version):
+        # This URL is computed pinging the place where the latest version is stored. To avoid slowdown
+        # due to repeated pinging, we store the URL in a private class attribute to do the job only once per version
+        openssl_urls = getattr(Openssl, '_openssl_url', {})
+        openssl_url = openssl_urls.get(version, None)
+        # Same idea, but just to avoid issuing the same message multiple times
+        warnings_given_to_user = getattr(Openssl, '_warnings_given', {})
+        if openssl_url is None:
+            latest = 'http://www.openssl.org/source/openssl-{version}.tar.gz'
+            older = 'http://www.openssl.org/source/old/{version_number}/openssl-{version_full}.tar.gz'
+            # Try to use the url where the latest tarballs are stored. If the url does not exist (404), then
+            # return the url for older format
+            version_number = '.'.join([str(x) for x in version[:-1]])
+            older_url = older.format(version_number=version_number, version_full=version)
+            latest_url = latest.format(version=version)
+            response = urllib.urlopen(latest.format(version=version))
+            if response.getcode() == 404:
+                openssl_url = older_url
+                # Checks if we already warned the user for this particular version of OpenSSL.
+                # If not we display a warning message and mark this version
+                if not warnings_given_to_user.get(version, False):
+                    tty.warn('This installation depends on an old version of OpenSSL, which may have known security issues. ')
```

Comment:
Versions which are in the `old` directory are likely to have known issues. I think the language should be stronger (or we should just ignore the old directory completely.